code/obj/machinery/dialysis.dm

```diff
@@ -0,0 +1,172 @@
+		..()
+		src.create_reagents(100)
+		if (!length(chem_whitelist))
+			CRASH("[src] tried to fetch the global chem whitelist but it has a length of 0!")
+		src.whitelist = chem_whitelist
+		src.UpdateOverlays(image(src.icon, "pump-off"), "pump")
+		src.UpdateOverlays(image(src.icon, "screen-off"), "screen")
+		src.UpdateOverlays(image(src.icon, "tubing"), "tubing")
+
+	disposing()
+		if (src.patient)
+			src.stop_dialysis()
+		..()
+
+	emag_act(mob/user, obj/item/card/emag/E)
+		if (src.hacked) return FALSE
+		src.hacked = TRUE
+		src.audible_message("<span class='game say'><span class='name'>[src]</span> beeps, \"Dialysis protocols inversed.\"")
+		logTheThing(LOG_ADMIN, user, "emagged [src] at [log_loc(user)].")
+		logTheThing(LOG_DIARY, user, "emagged [src] at [log_loc(user)].", "admin")
+		message_admins("[key_name(usr)] emagged [src] at [log_loc(user)].")
+
+	attack_hand(mob/user)
+		src.anchored = !src.anchored
+		boutput(user, "You [src.anchored ? "apply" : "release"] \the [src.name]'s brake.")
+
+	get_desc(dist, mob/user)
+		..()
+		if (src.hacked)
+			. += " Something about it seems a little off."
+		. += " Its internal blood reservoir is [get_fullness((src.reagents.total_volume / src.reagents.maximum_volume) * 100)]."
+
+	mouse_drop(atom/over_object as mob|obj)
+		var/mob/living/carbon/new_patient = over_object
+		var/mob/living/user = usr
+		if (isliving(user) && iscarbon(new_patient) && can_act(user) && in_interact_range(src, user) && in_interact_range(new_patient, user))
+			if (src.patient)
+				if (new_patient == src.patient)
+					new_patient.tri_message(user,\
+					"<span class='notice'><b>[user]</b> removes [src]'s cannulae from [new_patient == user ? "[his_or_her(new_patient)]" : "[new_patient]'s"] arm.</span>",\
+					"<span class='notice'>You remove [src]'s cannulae from [new_patient == user ? "your" : "[new_patient]'s"] arm.</span>",\
+					"<span class='notice'>[new_patient == user ? "You remove" : "<b>[user]</b> removes"] [src]'s cannulae from your arm.</span>")
+					return src.stop_dialysis()
+				else return boutput(user, "<span class='alert'>[src] already has a patient attached!</span>")
+			new_patient.tri_message(user,\
+			"<span class='notice'><b>[user]</b> begins inserting [src]'s cannulae into [new_patient == user ? "[his_or_her(new_patient)]" : "[new_patient]'s"] arm.</span>",\
+			"<span class='notice'>[new_patient == user ? "You begin" : "<b>[user]</b> begins"] inserting [src]'s cannulae into your arm.</span>",\
+			"<span class='notice'>You begin inserting [src]'s cannulae into [new_patient == user ? "your" : "[new_patient]'s"] arm.</span>")
+			logTheThing(LOG_COMBAT, user, "tries to hook up a dialysis machine [log_reagents(src)] to [constructTarget(new_patient,"combat")] at [log_loc(user)].")
+			SETUP_GENERIC_ACTIONBAR(user, src, 3 SECONDS, PROC_REF(cannulate), list(new_patient, user), src.icon, "dialysis-map", null, null)
+		..()
+
+	process(mult)
+		..()
+		if (!src.patient || !ishuman(src.patient))
```

Comment:
```suggestion
		if (!src.patient || !ishuman(src.patient) || QDELETED(src.patient))
```

Reply:
Though that's probably also caught in in_interact_range below. But idk makes more sense here.